solr/contrib/ltr/src/java/org/apache/solr/ltr/util/LTRRerankHelper.java

```diff
@@ -0,0 +1,134 @@
+  
+  public LTRRerankHelper(SolrQueryRequest request, SolrParams params){
+    this.request = request;
+    this.params = params;
+  }
+  
+  public ManagedModelStore getModelStore(){
+    return (ManagedModelStore) request.getCore().getRestManager()
+        .getManagedResource(CommonLTRParams.MODEL_STORE_END_POINT);
+  }
+  
+  public ManagedFeatureStore getFeatureStore(){
+    return (ManagedFeatureStore) request.getCore().getRestManager()
+        .getManagedResource(CommonLTRParams.FEATURE_STORE_END_POINT);
+  }
+  
+  
+  public LTRScoringAlgorithm getModel(String modelName){
+    ManagedModelStore modelStore = getModelStore();
+    final LTRScoringAlgorithm rerankModel = modelStore.getModel(modelName);
+    if (rerankModel == null) {
+      throw new SolrException(ErrorCode.BAD_REQUEST,
+          "cannot find " + CommonLTRParams.MODEL + " " + modelName);
+    }
+    return rerankModel;
+  }
+  
+  public String getDefaultFeatureStoreName(){
+   return CommonLTRParams.DEFAULT_FEATURE_STORE_NAME;
+  }
+  
+  
+  public String getStoreName(){
+    String store = params.get(CommonLTRParams.STORE);
+    return store;
+  }
+  
+  public String getModelName(){
+    String modelName = params.get(CommonLTRParams.MODEL);
+    if ((modelName == null) || modelName.isEmpty()) {
+      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+          "cannot find a model in the request");
+    }
+    return modelName;
+  }
+  
+  public String getStoreNameFromContext(){
+    String store = (String)request.getContext().get(CommonLTRParams.STORE);
+    return store;
+  }
+  
+  public ModelQuery getModelFromContext(){
+    return (ModelQuery)request.getContext().get(CommonLTRParams.MODEL);
+  }
+  
+  public void setModelInContext(ModelQuery reRankModel){
+    request.getContext().put(CommonLTRParams.MODEL, reRankModel);
+  }
+  
+  public boolean isLoggingFeatures(){
+    Boolean log = (Boolean)request.getContext().get(CommonLTRParams.LOG_FEATURES_QUERY_PARAM);
+    if (log == null) return true;
+    return log;
+  }
+  
+  public boolean hasModelInContext(){
+    return request.getContext().containsKey(CommonLTRParams.MODEL);
+  }
+  
+  public ModelQuery getRerankModel(LTRScoringAlgorithm model){
+    if (model == null) {
+      throw new SolrException(ErrorCode.BAD_REQUEST,
+          "invalid model");
+    }
+    final ModelQuery reRankModel = new ModelQuery(model);
+    
+    // External features
+    final Map<String,String> externalFeatureInfo = LTRUtils.extractEFIParams(params);
+    reRankModel.setExternalFeatureInfo(externalFeatureInfo);
+    // Request
+    reRankModel.setRequest(request);
+    return reRankModel;
+  }
+  
+  public ModelWeight getModelWeight(ModelQuery reRankModel, IndexSearcher searcher){
+    Weight w;
+    try {
+      w = searcher.createNormalizedWeight(reRankModel, true);
+    } catch (final IOException e) {
+      throw new SolrException(ErrorCode.BAD_REQUEST, e.getMessage(), e);
```

Comment:
We can get rid of this try catch, no need to rewrap an exception here since we dont really add anything